Jenkinsfile

```diff
@@ -36,6 +36,14 @@ pipeline {
       }
     }
 
+
+    stage("Lint Commits") {
+      steps {
+        sh 'git fetch --all'
+        sh 'npm run commitlint -- --from "\$(git ls-remote https://github.com/dcos/dcos-ui \"\$CHANGE_TARGET\" | head -1 | cut -f1)"'
```

Comment:
I wonder if using https://github.com/marionebl/commitlint/blob/master/docs/reference-api.md would be more maintainable than `head -1 | cut -f1`?

Reply:
The problem this `git ls-remote` subshell tries to solve is that the base branch is not known to git. Writing this in a JS script doesn't really help, it's still hacky because of the way Jenkins sets up the repository. 

The other option for us would be to `git clone` the repo in the beginning, which would leave us with the task to implement things like "merge and test" manually.

The "best" implementation would be this, if you know how to fix the error, please let me know: https://github.com/dcos/dcos-ui/pull/3424

Reply:
Do you thinks it's a good idea to lint messages _before_ building?

Reply:
I thought about this for a while and I came to the conclusion that making it fail early is a good approach. commit messages are easy and quick to fix, getting this feedback before more expensive tests run will improve the feedback time.

Reply:
Let's move it before and merge

Reply:
done